tests/aws/services/stepfunctions/v2/error_handling/test_states_errors.py

```diff
@@ -0,0 +1,176 @@
+import json
+
+from localstack_snapshot.snapshots.transformer import RegexTransformer
+
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from tests.aws.services.stepfunctions.templates.errorhandling.error_handling_templates import (
+    ErrorHandlingTemplate as EHT,
+)
+from tests.aws.services.stepfunctions.utils import create_and_record_execution
+
+
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..loggingConfiguration",
+        "$..tracingConfiguration",
+    ]
+)
+class TestStatesErrors:
+    @markers.aws.validated
+    def test_service_task_lambada_data_limit_exceeded_on_large_utf8_response(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        create_lambda_function,
+        sfn_snapshot,
+    ):
+        function_name = f"lambda_func_{short_uid()}"
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=EHT.LAMBDA_FUNC_LARGE_OUTPUT_STRING,
+            runtime="python3.9",
```

Comment:
Can we use the typed `Runtime.python3_12` here instead?

It would be great to adopt the latest Lambda runtimes for new tests because the al2023 image size is massively smaller, saving us pull time.

Reply:
same for the other new test cases

FYI: Bojan is currently updating the Lambda runtimes for existing tests, so no need to update them to prevent conflicts